mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -334,35 +408,8 @@ bool Conv2dGenerator::hasWorkspace(OpBuilder &builder) const {
     miopen::ConvOpType dir = config.operation.getValue();
     if ((dir == miopen::ConvOpType::BwdWeight) && config.xdlops &&
         (dataType == builder.getF16Type())) {
-
-      // Logic to check if we need to pad along GemmM/N/K dimension for backward
-      // weight convolution.
-      auto inDim = canonicalizeDims(config.inputDimension, config.inputLayout);
-      auto filDim =
-          canonicalizeDims(config.filterDimension, config.filterLayout);
-      auto outDim =
-          canonicalizeDims(config.outputDimension, config.outputLayout);
-
-      int64_t gemmMSize, gemmNSize, gemmKSize;
-      gemmMSize = filDim["k"];
-      gemmKSize = outDim["n"] * outDim["h"] * outDim["w"];
-      gemmNSize = filDim["c"] * filDim["y"] * filDim["x"];
```

Comment:
Can you go find all the instances of this out in `ConvToGemmLowering` and clean them up as well?

Reply:
Tracking it at https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/467 and I'll cover it in the next sprint.